src/components/Data/Documents/Page.vue

```diff
@@ -398,12 +472,111 @@ export default {
     },
     documents: {
       immediate: true,
+      deep: true,
       handler() {
         this.addHumanReadableDateFields()
       }
+    },
+    notifications: {
+      deep: true,
+      handler(value) {
+        if (value.length) {
+          this.$bvToast.show('realtime-notification-toast')
+        } else {
+          this.$bvToast.hide('realtime-notification-toast')
+        }
+      }
+    },
+    listViewType: {
+      immediate: true,
+      handler(value) {
+        if (value === 'list') {
+          this.$set(this, 'realtimeSettings', this.listViewRealtimeSettings)
+          this.enableRealtime = true
+        }
+        if (value === 'column') {
+          this.$set(this, 'realtimeSettings', this.columnViewRealtimeSettings)
+          this.enableRealtime = true
+        }
+        if (value === 'time-series') {
+          this.$set(
+            this,
+            'realtimeSettings',
+            this.timeSeriesViewRealtimeSettings
+          )
+          this.enableRealtime = false
+        }
+        if (value === 'map') {
+          this.$set(this, 'realtimeSettings', this.mapViewRealtimeSettings)
+          this.enableRealtime = false
+        }
+      }
     }
   },
   methods: {
+    async unsubscribeToCurrentDocs() {
+      if (this.subscribeRoomId) {
+        await this.$kuzzle.realtime.unsubscribe(this.subscribeRoomId)
+        this.subscribeRoomId = null
+      }
+    },
+    async subscribeToCurrentDocs() {
+      try {
+        await this.unsubscribeToCurrentDocs()
+        const roomId = await this.$kuzzle.realtime.subscribe(
+          this.indexName,
+          this.collectionName,
+          {},
+          this.realtimeNotifCallback
+        )
+        this.subscribeRoomId = roomId
+      } catch (error) {
+        this.$log.error(error)
+      }
+    },
+    formatMeta(_kuzzle_info) {
+      return {
+        author:
+          _kuzzle_info.author === '-1' ? 'Anonymous (-1)' : _kuzzle_info.author,
+        updater:
+          _kuzzle_info.updater === '-1'
+            ? 'Anonymous (-1)'
+            : _kuzzle_info.updater,
+        createdAt: _kuzzle_info.createdAt,
+        updatedAt: _kuzzle_info.updatedAt
+      }
+    },
+    applyNotification(notification) {
+      let documents = JSON.parse(JSON.stringify(this.documents))
+      const payload = {
+        _id: notification.result._id,
+        ...notification.result._source,
+        _kuzzle_info: notification.result._source._kuzzle_info
+          ? this.formatMeta(notification.result._source._kuzzle_info)
+          : undefined
+      }
+      if (['update', 'replace'].includes(notification.action)) {
+        const documentIdx = documents.findIndex(
+          doc => doc._id === notification.result._id
+        )
+
+        if (documentIdx !== -1) {
+          documents[documentIdx] = payload
+        }
+      } else if (notification.action === 'create') {
+        documents.push(payload)
```

Comment:
We should not push the new documents in the current document page, because this is a bit like trying to re-implement the logic of pagination on the frontend. This logic belongs to Elasticsearch and it's its own responsibility.